apollo-router/src/cache/redis.rs

```diff
@@ -657,29 +692,25 @@ impl RedisCacheStorage {
         ttl: Option<Duration>,
     ) {
         tracing::trace!("inserting into redis: {:#?}", data);
+        let expiration = ttl
+            .or(self.ttl)
+            .map(|ttl| Expiration::EX(ttl.as_secs() as i64));
 
-        let r = match ttl.as_ref().or(self.ttl.as_ref()) {
-            None => self.inner.mset(data.to_owned()).await,
-            Some(ttl) => {
-                let expiration = Some(Expiration::EX(ttl.as_secs() as i64));
-                let pipeline = self.inner.next().pipeline();
-
-                for (key, value) in data {
-                    let _ = pipeline
-                        .set::<(), _, _>(
-                            self.make_key(key.clone()),
-                            value.clone(),
-                            expiration.clone(),
-                            None,
-                            false,
-                        )
-                        .await;
-                }
+        // NB: if we were using MSET here, we'd need to split the keys by hash slot. however, fred
+        // seems to split the pipeline by hash slot in the background.
+        let pipeline = self.inner.next().pipeline();
+        for (key, value) in data {
+            let key = self.make_key(key.clone());
+            let _ = pipeline
+                .set::<(), _, _>(key, value.clone(), expiration.clone(), None, false)
+                .await;
+        }
 
-                pipeline.last().await
-            }
-        };
-        tracing::trace!("insert result {:?}", r);
+        let result: Result<Vec<()>, _> = pipeline.all().await;
```

Comment:
why switch from `pipeline.last()` to `pipeline.all()`? I don't think anything hinges on it, but not sure; `all()` returns an [array of all responses](https://docs.rs/fred/10.1.0/fred/clients/struct.Pipeline.html#method.all), `last()` just the [last](https://docs.rs/fred/10.1.0/fred/clients/struct.Pipeline.html#method.last) one

it looks like we're only using the `Err`s, so maybe `last()` is better to reduce memory waste?